src/webgl/3d_primitives.js

```diff
@@ -10,6 +10,179 @@ import p5 from '../core/main';
+ * }
+ * </code>
+ * </div>
+ *
+ * @alt
+ * A series of different flat, curved, and 3D shapes floating in space.
+ */
+p5.prototype.beginGeometry = function() {
+  return this._renderer.beginGeometry();
+};
+
+/**
+ * Finishes creating a new <a href="#/p5.Geometry">p5.Geometry</a> that was
+ * started using <a href="#/p5/beginGeometry">beginGeometry()</a>. One can also
+ * use <a href="#/p5/buildGeometry">buildGeometry()</a> to pass a function that
+ * draws shapes.
+ *
+ * @method endGeometry
+ * @returns {p5.Geometry} The model that was built.
+ */
+p5.prototype.endGeometry = function() {
+  return this._renderer.endGeometry();
+};
+
+/**
+ * Creates a new <a href="#/p5.Geometry">p5.Geometry</a> that contains all
+ * the shapes drawn in a provided callback function. The returned combined shape
+ * can then be drawn all at once using <a href="#/p5/model">model()</a>.
+ *
+ * If you need to draw complex shapes every frame which don't change over time,
+ * combining them with `buildGeometry()` once and then drawing that will likely
+ * run faster than repeatedly drawing the individual pieces.
+ *
+ * One can also draw shapes directly between
+ * <a href="#/p5/beginGeometry">beginGeometry()</a> and
+ * <a href="#/p5/endGeometry">endGeometry()</a> instead of using a callback
+ * function.
+ *
+ * @method buildGeometry
+ * @param {Function} callback A function that draws shapes.
+ * @returns {p5.Geometry} The model that was built from the callback function.
+ *
+ * @example
+ * <div>
+ * <code>
+ * let tree;
+ * let button;
+ *
+ * function setup() {
+ *   createCanvas(100, 100, WEBGL);
+ *   button = createButton('New');
+ *   button.mousePressed(makeTree);
+ *   makeTree();
+ * }
+ *
+ * function makeTree() {
+ *   if (tree) freeGeometry(tree);
+ *   tree = buildGeometry(() => {
+ *     const addBranch = function(depth) {
+ *       push();
+ *       translate(0, -50);
+ *       cylinder(15, 100);
+ *       translate(0, -50);
+ *       if (depth >= 5) {
+ *         sphere(30);
+ *       } else {
+ *         const numChildren = round(random(1, 3));
+ *         for (let i = 0; i < numChildren; i++) {
+ *           push();
+ *           rotateZ(random(-0.3, 0.3) * PI);
+ *           rotateY(random(TWO_PI));
+ *           addBranch(depth + 1);
+ *           pop();
+ *         }
+ *       }
+ *       pop();
+ *     };
+ *     translate(0, 25);
+ *     scale(0.18);
+ *     addBranch(0);
+ *   });
+ * }
+ *
+ * function draw() {
+ *   background(255);
+ *   lights();
+ *   noStroke();
+ *   orbitControl();
+ *   model(tree);
+ * }
+ * </code>
+ * </div>
+ *
+ * @alt
+ * A 3D tree made of multiple cylinders and spheres.
+ */
+p5.prototype.buildGeometry = function(callback) {
+  return this._renderer.buildGeometry(callback);
+};
+
+/**
+ * Clears the resources of a model to free up browser memory. A model whose
+ * resources have been cleared can still be drawn, but the first time it is
+ * drawn again, it might take longer.
+ *
+ * @method freeGeometry
```

Comment:
Is `free` the correct name? Do we use this anywhere else in p5? Alternative could be `release`. Not really preferential to either but worth a discussion.

Reply:
The closest comparison I can think of is `remove()` for graphics and framebuffers. This is a little different, because it only clears the resources from one rendering context (e.g. if you use a model and draw it to multiple graphics, you can clear each webgl context's data separately), and the CPU bound data is only gotten rid of when it's garbage collected away, so I think `remove()` is probably not right here.

I don't have a preference between `free` or `release` or something else of that sort though! Whichever we think sounds clearest, since we're setting a new precedent.

Reply:
We had an internal `_freeBuffers` before, which was private, so this initial name was just taking that and making a public version

Reply:
free sounds good to me, just wanted to raise the issue on the chance that this language was used elsewhere.